File: web/src/data/NoteStats.ts
import { Timestamp } from 'firebase/firestore'

/** IMPORTANT: Keep in sync with Kotlin NoteStats.kt and docs/schema.md */
export interface NoteStats {
  lastAccessedAt: Timestamp | null
  viewCount: number
  /** YYYY-MM-DD keys in the device's local timezone, so a midnight crossing reads as the user-perceived calendar day. */
  viewedDays: Record<string, boolean>
}

export function noteStatsFromFirestore(data: Record<string, unknown>): NoteStats {
  return {
    lastAccessedAt: (data.lastAccessedAt as Timestamp) ?? null,
    viewCount: (data.viewCount as number) ?? 0,
    viewedDays: (data.viewedDays as Record<string, boolean>) ?? {},
  }
}

File: app/src/main/java/org/alkaline/taskbrain/data/NoteStats.kt
package org.alkaline.taskbrain.data

import com.google.firebase.Timestamp

/** IMPORTANT: Keep in sync with docs/schema.md and web NoteStats.ts */
data class NoteStats(
    val lastAccessedAt: Timestamp? = null,
    val viewCount: Int = 0,
    /** YYYY-MM-DD keys in the device's local timezone, so a midnight crossing reads as the user-perceived calendar day. */
    val viewedDays: Map<String, Boolean> = emptyMap(),
)
